autoAdjustHwloc.hpp

```diff
@@ -0,0 +1,211 @@
+#pragma once
+
+#include "console.h"
+#include <hwloc.h>
+
+/*
+ * compat_popcnt is adapted from public domain code by:
+ * Portable Snippets - https://gitub.com/nemequ/portable-snippets
+ * Created by Evan Nemerson <evan@nemerson.com>
+ */
+int compat_popcnt(size_t x) 
+{
+	x = x - ((x >> 1) & (size_t)~(size_t)0/3);
```

Comment:
Instead of adding this little magic code we can also add a for loop what is human readable.
The code part is not time critical.
```C++
template< typename T >
int compat_popcnt( T x ) 
{
  size_t count = 0;
  for( int i=0; i < sizeof(T) - 1; ++i )
    if( ((x>>i) & 1) == 1 )
      ++count;
}
```